src/main/scala/com/github/sashjakk/http/Codecs.scala

```diff
@@ -0,0 +1,48 @@
+package com.github.sashjakk.http
+
+import cats.implicits.catsSyntaxEitherId
+import com.github.sashjakk.interval.Interval
+import com.github.sashjakk.spot.book.{Booking, BookingCreate}
+import com.github.sashjakk.spot.share.{Share, ShareCreate}
+import com.github.sashjakk.spot.{FreeSpot, Spot, SpotCreate}
+import com.github.sashjakk.user.{User, UserCreate}
+import io.circe._
+import io.circe.generic.semiauto.{deriveCodec, deriveEncoder}
+import io.circe.syntax.EncoderOps
+
+import java.time.Instant
+
+object Codecs {
+  implicit val restApiErrorDecoder: Encoder[APIError] = deriveEncoder
+
+  implicit val userCreateCodec: Codec[UserCreate] = deriveCodec
+  implicit val userCodec: Codec[User] = deriveCodec
+
+  implicit val spotCreateDecoder: Codec[SpotCreate] = deriveCodec
+  implicit val spotEncoder: Codec[Spot] = deriveCodec
+
+  implicit val shareCreateCodec: Codec[ShareCreate] = deriveCodec
+  implicit val shareEncoder: Codec[Share] = deriveCodec
+
+  implicit val intervalEncoder: Encoder[Interval] =
+    interval => Json.obj(("from", interval.from.asJson), ("to", interval.to.asJson))
+  implicit val intervalDecoder: Decoder[Interval] =
+    cursor =>
+      for {
+        from <- cursor.downField("from").as[Instant]
+        to <- cursor.downField("to").as[Instant]
+        interval <- {
+          type ErrOr[A] = Either[Throwable, A]
+
+          Interval.from[ErrOr](from, to) match {
```

Comment:
Is it possible to use `Decoder.Result` directly? Cause it's also typealias for Either

Reply:
Yes, you can use it for everything that implements MonadThrow (you actually need only ApplicativeThrow).

Reply:
not solved 🙈 will continue like this for now, todo left